frontend/src/js/components/PageViewer/PageCache.ts

```diff
@@ -119,6 +130,40 @@ export class PageCache {
     };
   };
 
+  refreshPreview = (pageNumber: number, preview: Promise<CachedPreview>, containerSize: number): CachedPage => {  
+    this.setContainerSize(containerSize);
+    const originalPreviewData = this.previewCache.get(pageNumber);
+    const newPreview = updatePreview(preview, containerSize);
+    const newPreviewCache = {
+      previewAbortController: originalPreviewData.previewAbortController,
+      preview: newPreview,
+      previewContainerSize: this.containerSize    
+    };  
+
+    const data = this.dataCache.get(pageNumber);
+
+    this.previewCache.replace(pageNumber, newPreviewCache);
+
+    const newPreviewData = this.previewCache.get(pageNumber);
+
+    return {
+      ...newPreviewData,
+      ...data,
+    };
+  };
+
+  getPageAndRefreshPreview = (pageNumber: number): CachedPage => {
```

Comment:
Seems like this one is unused now, can we get rid of it?

Reply:
👍